src/hooks/project-hooks.tsx

```diff
@@ -108,14 +138,24 @@ export const ProjectProvider = ({
       type: "edit-in-makecode",
     });
     await doAfterEditorUpdate(() => {
-      setEditorOpen(true);
+      navigate(createCodePageUrl());
       return Promise.resolve();
     });
-  }, [doAfterEditorUpdate, logging, setEditorOpen]);
-
+  }, [doAfterEditorUpdate, logging, navigate]);
+  const browserNavigationToEditor = useCallback(async () => {
+    try {
+      await doAfterEditorUpdate(() => {
+        return Promise.resolve();
+      });
+      return true;
+    } catch (e) {
+      // doAfterEditorUpdate only fails when navigating to the code page directly.
+      // In this case, the caller of browserNavigationToEditor redirects.
+      return false;
```

Comment:
We could now check for the right CodeEditorError here, but it's less clear what to do if it's an unexpected error. Navigate to a different page anyway, or possibly reload the page.